software/base/src/main/java/brooklyn/entity/software/MachineLifecycleEffectorTasks.java

```diff
@@ -561,7 +560,7 @@ public void stop(ConfigBag parameters) {
 
         boolean isEntityStopped = entity().getAttribute(SoftwareProcess.SERVICE_STATE_ACTUAL)==Lifecycle.STOPPED;
 
-        DynamicTasks.queue("pre-stop", new Callable<String>() { public String call() {
+        DynamicTasks.queue("pre-stop of " + entity() + ", submitted by " + Tasks.current(), new Callable<String>() { public String call() {
```

Comment:
It was helpful to discover what was being queued from where when a bug in Clocker caused massive task recursion. Hundreds of "pre-stops" in the logs were no help, the message above was. See [this comment](https://github.com/brooklyncentral/clocker/pull/91#issue-55657091) for more.